File: api/src/main/java/com/careerlabs/lms/api/student/service/StudentService.java
package com.careerlabs.lms.api.student.service;

import com.careerlabs.lms.api.student.dto.request.StudentCreateRequest;
import com.careerlabs.lms.api.student.dto.request.StudentUpdateRequest;
import com.careerlabs.lms.api.student.dto.response.StudentCountResponse;
import com.careerlabs.lms.api.student.dto.response.StudentPageResponse;
import com.careerlabs.lms.api.student.dto.response.StudentResponse;
import com.careerlabs.lms.api.student.entity.PlacementStatus;

import com.careerlabs.lms.api.student.dto.response.StudentBulkImportResponse;
import org.springframework.web.multipart.MultipartFile;

public interface StudentService {

    StudentPageResponse list(String search, Long batchId, String status, PlacementStatus placementStatus,
                              int page, int limit);

    StudentCountResponse count();

    StudentResponse get(Long id);

    StudentResponse create(StudentCreateRequest request);

    StudentResponse update(Long id, StudentUpdateRequest request);

    StudentResponse toggleStatus(Long id);

    StudentResponse assignToBatch(Long studentId, Long batchId);

    void delete(Long id);

    StudentBulkImportResponse bulkImport(MultipartFile file, boolean enrollImportedStudents,
                                         Long defaultCourseId, Long defaultBatchId);
}

File: frontend/src/services/studentService.js
import apiCall from '@/utilities/apiCall'

const studentService = {
  list: (params) => apiCall({ method: 'GET', url: '/students', params }),

  count: () => apiCall({ method: 'GET', url: '/students/count' }),

  get: (id) => apiCall({ method: 'GET', url: `/students/${id}` }),

  create: (student) => apiCall({ method: 'POST', url: '/students', data: student }),

  update: (id, student) => apiCall({ method: 'PUT', url: `/students/${id}`, data: student }),

  updatePlacementStatus: (id, placementStatus) => apiCall({
    method: 'PATCH',
    url: `/students/${id}/placement-status`,
    data: { placementStatus },
  }),

  toggleStatus: (id) => apiCall({ method: 'PATCH', url: `/students/${id}/status` }),

  remove: (id) => apiCall({ method: 'DELETE', url: `/students/${id}` }),

  bulkImport: ({ file, enrollImportedStudents, defaultCourseId, defaultBatchId }) => {
    const formData = new FormData()
    formData.append('file', file)
    formData.append('enrollImportedStudents', enrollImportedStudents ? 'true' : 'false')
    if (defaultCourseId) formData.append('defaultCourseId', defaultCourseId)
    if (defaultBatchId) formData.append('defaultBatchId', defaultBatchId)
    return apiCall({
      method: 'POST',
      url: '/students/bulk-import',
      data: formData,
    })
  },
}

export default studentService
